From d7d4c26fd27066c40caf60ec1b5e3404cdac208b Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Tue, 12 Dec 2017 05:04:35 -0600
Subject: [PATCH] "0001-libutils-makeup-threadname-if-one-is-not-sent
 _______________________________

*Previous From 515668b788d04996fb48399228bd85ad793a9126 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sun, 26 Mar 2017 11:28:07 -0500
*Previous Subject: [PATCH] libutils: makeup threadname if one is not sent
*
*Change-Id: Ia35caaa415a4e89a621a0ef867e05df463f6967f"
---
 libutils/Android.bp  |  1 -
 libutils/Threads.cpp | 12 ++++++++----
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/libutils/Android.bp b/libutils/Android.bp
index 5e76279..14c18ae 100644
--- a/libutils/Android.bp
+++ b/libutils/Android.bp
@@ -64,7 +64,6 @@ cc_library {
         "misc.cpp",
     ],
 
-    cflags: ["-Werror"],
     include_dirs: ["external/safe-iop/include"],
     header_libs: [
         "libutils_headers",
diff --git a/libutils/Threads.cpp b/libutils/Threads.cpp
index 6317c32..6ac1a57 100644
--- a/libutils/Threads.cpp
+++ b/libutils/Threads.cpp
@@ -661,8 +661,12 @@ status_t Thread::readyToRun()
 
 status_t Thread::run(const char* name, int32_t priority, size_t stack)
 {
-    LOG_ALWAYS_FATAL_IF(name == nullptr, "thread name not provided to Thread::run");
-
+    //LOG_ALWAYS_FATAL_IF(name == nullptr, "thread name not provided to Thread::run");
+    char nname[255];
+    if(name == nullptr)
+	sprintf(nname,"meticulus_%d",rand());
+    else
+	sprintf(nname,"%s",name);
     Mutex::Autolock _l(mLock);
 
     if (mRunning) {
@@ -684,10 +688,10 @@ status_t Thread::run(const char* name, int32_t priority, size_t stack)
     bool res;
     if (mCanCallJava) {
         res = createThreadEtc(_threadLoop,
-                this, name, priority, stack, &mThread);
+                this, nname, priority, stack, &mThread);
     } else {
         res = androidCreateRawThreadEtc(_threadLoop,
-                this, name, priority, stack, &mThread);
+                this, nname, priority, stack, &mThread);
     }
 
     if (res == false) {
-- 
2.7.4

